SPECS/cert-manager/cert-manager.spec

```diff
@@ -1,96 +1,117 @@
+
+%description controller
+cert-manager is a Kubernetes addon to automate the management and issuance of
+TLS certificates from various issuing sources.
+
+%package cmctl
+Summary: cert-manager's cmctl binary
+
+%description cmctl
+cmctl is a CLI tool manage and configure cert-manager resources for Kubernetes
+
+%package webhook
+Summary: cert-manager's webhook binary
+
+%description webhook
+Webhook component providing API validation, mutation and conversion functionality for cert-manager.
+
 %prep
 %autosetup -p1
 %setup -q -T -D -a 1
 
-%build
-export GO111MODULE=off
-mkdir -p %{_topdir}/go
-export GOPATH=%{_topdir}/go
-pushd $GOPATH
-tar -xvf %{SOURCE2} --strip-components=1 --no-same-owner
-popd
-
-export GO_REPOSITORY_USE_HOST_CACHE=1
-
-git config --global user.email you@example.com
-git config --global user.name "Your Name"
-git init
-git add .
-GIT_AUTHOR_DATE=2000-01-01T01:01:01 GIT_COMMITTER_DATE=2000-01-01T01:01:01 \
-git commit -m "Dummy commit just to satisfy bazel" &> /dev/null
-
-for cmd in cmd/* ; do
-  if [ "$cmd" != cmd/util ]; then
-    bazel --batch build --repository_cache=BAZEL_CACHE //$cmd
-  fi
-done
+go build -o bin/acmesolver cmd/acmesolver/main.go
+go build -o bin/cainjector cmd/cainjector/main.go
+go build -o bin/controller cmd/controller/main.go
+go build -o bin/cmctl cmd/ctl/main.go
+go build -o bin/webhook cmd/webhook/main.go
 
 %install
 mkdir -p %{buildroot}%{_bindir}
-%ifarch aarch64
-install -D -m0755 bazel-out/aarch64-fastbuild-ST-4c64f0b3d5c7/bin/cmd/ctl/kubectl-cert_manager %{buildroot}%{_bindir}/
-%else
-install -D -m0755 bazel-out/k8-fastbuild-ST-4c64f0b3d5c7/bin/cmd/ctl/kubectl-cert_manager %{buildroot}%{_bindir}/
-%endif
-install -D -m0755 bazel-bin/cmd/webhook/webhook_/webhook %{buildroot}%{_bindir}/
-install -D -m0755 bazel-bin/cmd/controller/controller_/controller %{buildroot}%{_bindir}/
-install -D -m0755 bazel-bin/cmd/cainjector/cainjector_/cainjector %{buildroot}%{_bindir}/
-install -D -m0755 bazel-bin/cmd/acmesolver/acmesolver_/acmesolver %{buildroot}%{_bindir}/
-
-%files
+install -D -m0755 bin/acmesolver %{buildroot}%{_bindir}/
+install -D -m0755 bin/cainjector %{buildroot}%{_bindir}/
+install -D -m0755 bin/controller %{buildroot}%{_bindir}/
+install -D -m0755 bin/cmctl %{buildroot}%{_bindir}/
+install -D -m0755 bin/webhook %{buildroot}%{_bindir}/
+
+%files acmesolver
+%license LICENSE
+%license LICENSES
+%doc README.md
+%{_bindir}/acmesolver
+
+%files cainjector
+%license LICENSE
+%license LICENSES
+%doc README.md
+%{_bindir}/cainjector
+
+%files controller
 %license LICENSE
+%license LICENSES
 %doc README.md
-%{_bindir}/*
+%{_bindir}/controller
+
+%files cmctl
+%license LICENSE
+%license LICENSES
+%doc README.md
+%{_bindir}/cmctl
+
+%files webhook
+%license LICENSE
+%license LICENSES
+%doc README.md
+%{_bindir}/webhook
 
 %changelog
+* Fri Aug 05 2022 Chris Gunn <chrisgun@microsoft.com> - 1.7.3-1
+- Update to v1.7.3
+- Split binaries into separate packages.
```

Comment:
That's great but for backward compatibility, let's keep the default package installing all of the binaries. This can be done with an empty `%files` section and a bunch of `Requires: %{name}-[subpackage_name]` under the `BuildRequires`.

Reply:
Fixed.